theano/sandbox/cuda/__init__.py

```diff
@@ -362,6 +362,9 @@ def c_libraries(self):
     def c_lib_dirs(self):
         return [config.dnn.library_path]
 
+    def c_compile_args(self):
+        return ['-rpath', config.dnn.library_path]
```

Comment:
Also, I just saw that you propose adding a space which is also a really bad idea which complicates lots of things (such as -Xcompiler processing in nvcc_compiler that will break if we do this).